src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -269,10 +269,35 @@ int sd_dhcp_client_set_mac(
                 const uint8_t *hw_addr,
                 const uint8_t *bcast_addr,
                 size_t addr_len,
-                uint16_t arp_type) {
+                unsigned short iftype) {
 
         assert_return(client, -EINVAL);
         assert_return(!sd_dhcp_client_is_running(client), -EBUSY);
+        assert_return(IN_SET(iftype, ARPHRD_ETHER, ARPHRD_INFINIBAND, ARPHRD_RAWIP, ARPHRD_NONE), -EINVAL);
```

Comment:
hmm, one more request: can you add a brief comment here that explains why ARPHRD_RAWIP and ARPHRD_NONE are listed here?

I think the other two are somewhat self-explanatory for the casual reader. But the two you are adding here deserve some brief explanation, i.e. mention which subsystems/devices use these, and how we should treat them.

Reply:
Sure.  I put the comment a couple of lines below in the switch statement, and also added a brief explanatory comment along similar lines in networkd-dhcp-common.c